src/main/java/cubicchunks/worldgen/GeneratorPipeline.java

```diff
@@ -70,9 +79,38 @@ public void checkStages() {
 	}
 
 	public void generate(Cube cube) {
-		GeneratorStage stage = cube.getGeneratorStage();
-		if (!stage.isLastStage()) {
-			stage.getProcessor().processor.add(cube.getAddress());
+		this.generate(cube, GeneratorStage.LIVE);
+	}
+
+	public void resume(Cube cube) {
+		cube.getCurrentStage().getProcessor().processor.add(cube.getAddress());
+	}
+
+	public void generate(Cube cube, GeneratorStage targetStage) {
+		GeneratorStage stage = cube.getCurrentStage();
+		
+		// Make sure the proper target stage is set.
+		if (cube.getTargetStage().precedes(targetStage)) {
+			cube.setTargetStage(targetStage);
+		}
+		
+		// If the cube has not yet reached the target stage, queue it for processing.
+		if (cube.getCurrentStage().precedes(cube.getTargetStage())) {
+			
+			// Register dependencies.
+			if (stage instanceof DependencyProvider) {
```

Comment:
It's not for performance. It just seems like a bad design. If most stages are expected to have dependencies, then making GeneratorStage abstract class that implements DependencyProvider may be a good idea.